### PR TITLE
Various fixes

### LINKED ISSUE #258: operation.mbr() ignore blockers
No more blockers in zoom calculations.

### PR DESCRIPTION
**Pull request by le-jeu**
marker completed by
merge phrasing + auto merge

fix #258

### CODE REVIEW

**Review by cloudkucooland — COMMENTED** [review]
looks fine on first read. I will merge into dev and let others test.

File: src/code/server.js [hunk 1]
```diff
@@ -22,6 +22,8 @@ export async function uploadOpPromise() {
   const newme = new WasabeeMe(response);
   newme.store();
   const newop = await opPromise(operation.ID);
+  // merge blockers and related portals
+  newop.mergeBlockers(operation);
```
Comment (cloudkucooland) [inline src/code/server.js:26]: I like this solution.

File: src/code/uiCommands.js [hunk 2]
```diff
@@ -388,6 +388,44 @@ export function zoomToOperation(operation) {
   }
 }
 
+export async function updateLocalOp(local, remote) {
+  const so = getSelectedOperation();
+  if (!local || !local.localchanged) {
+    // merge blockers and related portals
+    if (local) remote.mergeBlockers(local);
```
Comment (cloudkucooland) [inline src/code/uiCommands.js:395]: is this line redundant with 393? we can't be !local and local

File: src/code/marker.js [hunk 3]
```diff
@@ -120,10 +120,15 @@ export default class WasabeeMarker {
         console.error(err);
       }
     }
-    if (this.state == "completed" && this.completedBy) {
-      assignment.innerHTML = wX("COMPLETED BY", {
-        agentName: this.completedBy,
-      });
+    if (this.state == "completed" && this.completedID) {
```
Comment (cloudkucooland) [inline src/code/marker.js:123]: thank you for catching this... I thought I'd already done it

File: src/code/marker.js [hunk 4]
```diff
@@ -20,7 +20,7 @@ export default class WasabeeMarker {
     this.portalId = obj.portalId;
     this.type = obj.type;
     this.comment = obj.comment ? obj.comment : ""; // why "" and not null? This isn't go
-    this.completedBy = obj.completedBy ? obj.completedBy : "";
+    this.completedID = obj.completedID ? obj.completedID : "";
```
Comment (cloudkucooland) [inline src/code/marker.js:23]: did my commits on this get lost?!

**Review by cloudkucooland — APPROVED** [review]
looks good to me. I'll pull the recent changes and push to dev

File: src/code/uiCommands.js (hunk 2 above)
Comment (le-jeu) [reply src/code/uiCommands.js:395]: it can be `local` and `!local.localchanged`

File: src/code/operation.js [hunk 5]
```diff
@@ -1287,6 +1287,21 @@ export default class WasabeeOp {
     for (const b of op.blockers) this.blockers.push(b); // do not use addBlocker
   }
 
+  mergeZones(op) {
```
Comment (cloudkucooland) [inline src/code/operation.js:1290]: does this account for zone deletion?

Comment (cloudkucooland) [reply src/code/operation.js:1290]: does it need to?

Comment (le-jeu) [reply src/code/operation.js:1298]: no, it's more an "import" than a "merge"
it's there mainly for the team import

**Pull request merged**